cpp/CMakeLists.txt

```diff
@@ -194,16 +194,15 @@ static|shared (default shared)")
   # * SYSTEM: Use CMake's find_package and find_library without any custom
   #   paths. If packages are on non-default locations, let the user indicate it
   #   from the outside using the *_ROOT variables.
-  #   If your system packages are in general in a non-default location, you can
-  #   set a general default for the *_ROOT variables using ARROW_PACKAGE_PREFIX.
-  # * CONDA: Same as system but set all *_ROOT variables to ENV{CONDA_PREFIX}.
+  #   If your system packages are in general in a non-default location, or if
+  #   you are using a non-standard toolchain, you can set a general default for
+  #   the *_ROOT variables using ARROW_PACKAGE_PREFIX.
+  # * CONDA: Same as system but set all *_ROOT variables to
+  #   ENV{CONDA_PREFIX}. If this is run within an active conda environment,
+  #   then ENV{CONDA_PREFIX} will be used for dependencies unless
+  #   ARROW_DEPENDENCY_SOURCE is set explicitly to one of the other options
   # * BREW: Use SYSTEM but search for select packages with brew.
-  if(NOT "$ENV{CONDA_PREFIX}" STREQUAL "")
```

Comment:
I would actually like to keep this as the main strategy. Having conda packages and the build toolchain split is something that I would only advise to expert users since the compiler migration. With the new toolchain using system compilers, binutils, … is going to be a pain unless one is on a brand new system. I rather have CMake error out on missing conda packages than weird compilers as the binutils/compiler/ABI is not matching.

Reply:
Alright, I can write `-DARROW_DEPENDENCY_SOURCE=SYSTEM -DARROW_PACKAGE_PREFIX=$TOOLCHAIN` in the meantime